packages/next/src/server/require.ts

```diff
@@ -29,10 +29,10 @@ const pagePathCache =
 export function getMaybePagePath(
   page: string,
   distDir: string,
-  locales?: string[],
-  appDirEnabled?: boolean
+  locales: string[] | undefined,
+  isAppPath: boolean
 ): string | null {
-  const cacheKey = `${page}:${distDir}:${locales}`
+  const cacheKey = `${page}:${distDir}:${locales}:${isAppPath}`
```

Comment:
Cool, I've seen this cache key but didn't realize it could also conflict.
Nice :+1: